source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1101,30 +1102,27 @@ updateRSFromPrimary
         checkIfHasPrimary()
         return
 
-    if description.setVersion is not null and description.electionId is not null:
+    if description.electionId is not null and description.setVersion is not null:
         # Election ids are ObjectIds, see
         # "using setVersion and electionId to detect stale primaries"
         # for comparison rules.
-        if (topologyDescription.maxSetVersion is not null and
-            topologyDescription.maxElectionId is not null and (
-                topologyDescription.maxSetVersion > description.setVersion or (
-                    topologyDescription.maxSetVersion == description.setVersion and
-                    topologyDescription.maxElectionId > description.electionId
-                )
-            ):
 
+        # Null values are always considered "less than"
```

Comment:
This comment seems unnecessary since null values are never compared.  There is a check on line 1105 for null in the description and the conditionals below guard against null for the max values.

Reply:
I was thinking that this was an important note as for why the ternary would return false or true.

Like:
```
maxSetVersionIsLess = topologyDescription.maxSetVersion < serverDescription.setVersion if topologyDescription.maxSetVersion is not null else true
# maxSetVersion is automatically less when it is null (that's the "else true"), otherwise get the result of the actual comparison
```

Reply:
Oh, I see.  That makes sense, then.